File: storage/db-core/src/main/java/com/coffee/gu/menu/MenuCategoryEntity.java
package com.coffee.gu.menu;

import com.coffee.gu.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

@Table(name = "menu_category")
@Entity
public class MenuCategoryEntity extends BaseEntity {
    private Long menuId;
    private Long categoryId;

    protected MenuCategoryEntity() {}

    public MenuCategoryEntity(Long menuId, Long categoryId) {
        this.menuId = menuId;
        this.categoryId = categoryId;
    }

    public MenuCategory toModel() {
        return new MenuCategory(id, menuId, categoryId);
    }

    public Long getMenuId() {return menuId;}

    public Long getCategoryId() {
        return categoryId;
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/menu/MenuCategoryEntity.kt
package com.coffee.gu.menu

import com.coffee.gu.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Table(name = "menu_category")
@Entity
class MenuCategoryEntity @JvmOverloads constructor(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    val menuId: Long,
    val categoryId: Long,
) : BaseEntity() {

    fun toModel(): MenuCategory {
        return MenuCategory(id, menuId, categoryId)
    }
}
